createFolders.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+PROJECT_NAME=data-and-algorithms
+DIRECTORY=${PWD}
```

Comment:
you can avoid most of the boilerplate and validation by using

```shell
DIRECTORY="$(readlink -f -- "$(dirname -- "$0")")"
```

yes, all those quotes are needed. otherwise you'll break people who clone this repository in a path that has spaces!